### PR TITLE
Upgrade our Google Analytics wrapper

### PR DESCRIPTION
**Pull request by chrsmith**
This PR upgrades the codepath we use to report data to Google Analytics. The functionality is pretty much the same, but we are now using the same format as the Google Cloud Tools for Visual Studio extension. The end result is that by reporting our data slightly different, it will be picked up by internal dashboards, etc.

### CODE REVIEW

**Comment by chrsmith** [comment]
@ivannaranjo PTAL for all the Google Analytics bits.
@ILMTitan FYI

File: Google.PowerShell/Common/ReportCmdletResults.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        /// Category for all analytic events reported through this instance.
+        /// </summary>
+        private AnalyticsEventCategory _category = AnalyticsEventCategory.CmdletInvocation;
+
+        public GoogleAnalyticsCmdletReporter(string clientID, AnalyticsEventCategory analyticsCategory)
+        {
+            // We report results to Google Analytics if the user has opted-into Cloud SDK metric reporting. However, we
+            // don't want to report data during development. (e.g. when running unit tests.)
+            string envVar = Environment.GetEnvironmentVariable("DISABLE_POWERSHELL_ANALYTICS");
+            bool debugFlag = (envVar == "TRUE");
```
Comment (ivannaranjo) [inline Google.PowerShell/Common/ReportCmdletResults.cs]: Suggestion, maybe less stringent check is better. Just check that is not empty. Checking for only TRUE can make it a bit error prone.

File: Google.PowerShell/Common/ReportCmdletResults.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        }
+
+        public void ReportFailure(string cmdletName, string parameterSet, int errorCode)
+        {
+            Report(_category, cmdletName, parameterSet, errorCode);
+        }
+
+        private void Report(AnalyticsEventCategory category, string cmdletName, string parameterSet, int? errorCode)
+        {
+            _reporter.Value?.ReportEvent(
```
Comment (ivannaranjo) [inline Google.PowerShell/Common/ReportCmdletResults.cs]: You are supposed to use the `ReportPageView()` method instead, analytics are reported as virtual page views with more information than events can handle.

File: Google.PowerShell/Common/GoogleAnalyticsReporter.cs [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            return result;
+        }
+
+        /// <summary>
+        /// Sends the hit data to the server.
+        /// </summary>
+        /// <param name="hitData">The hit data to be sent.</param>
+        private async void SendHitData(Dictionary<string, string> hitData)
+        {
+            using (var client = new HttpClient())
```
Comment (quoctruong) [inline Google.PowerShell/Common/GoogleAnalyticsReporter.cs]: I think HttpClient is supposed to be reused (https://www.asp.net/web-api/overview/advanced/calling-a-web-api-from-a-net-client) so maybe we shouldn't create a new instance of HttpClient everytime.

**Comment by chrsmith** [comment]
Abandoning the PR. Need to branch from a different commit of the VS plug-in code.

**Pull request closed without merging**